File: 백준/Silver/17103. 골드바흐 파티션/골드바흐 파티션.py
import sys, math
input = sys.stdin.readline


def prime_list(n):
    arr = [True] * (n+1)

    for i in range(2, n+1):
        if arr[i]:
            for j in range(i+i, n+1, i):
                arr[j] = False

    return arr

T = int(input())
prime = prime_list(1000000)
prime[0], prime[1] = False, False

for _ in range(T):
    x = int(input())
    cnt = 0
    for i in range(2, x):
        if prime[i] and prime[x-i]:
            cnt += 1

    print(math.ceil(cnt/2))

File: 백준/Silver/17103. 골드바흐 파티션/골드바흐 파티션.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split('\n')
input = input.map(x => parseInt(x))

function prime(x) {
    arr = new Array(x).fill(true)
    for (let i = 2; i < x; i++) {
        if (arr[i]) {
            for (let j = i+i; j < x; j += i) {
                arr[j] = false
            }
        }
    }
    return arr
}

const p = prime(1000000)
p[0] = false
p[1] = false
input.forEach((x, i) => {
    if (i !== 0) {
        let cnt = 0
        for (let j = 2; j < x; j++) {
            if (p[j] && p[x-j]) {
                cnt += 1
            }
        }
        console.log(Math.ceil(cnt/2))
}})
